microservices.yml

```diff
@@ -96,6 +96,16 @@ commands:
           name: Run unit tests in docker container
           command: |
             make ci-test-unit
+      - run:
```

Comment:
i think that this part (`cucumber-junit`) should be in the `makefile`, what was your reasoning for doing it here?

Reply:
@misiek-sf I disagree. I purposely added it here because its only purpose is to transform specifically to a format that CircleCI wants (and then send it to CircleCI right afterwards).

It is not used by services for their orchestration nor should the services be aware that the orb is sending their generic test report to CircleCI in a specific format that only the CI build env uses/needs.

Considering that responsibility tree, these two lines should remain in the orb.

Reply:
good point didn't think of it like that... but then why "mount" json file in the makefile then? maybe add your changes to `DOCKER_RUN_OPTS ` here instead of `makefile`?

Reply:
If I understand you correctly, I think you're asking why I've added this line in API repo:
https://github.com/MGMResorts/api/pull/92/files#diff-2f62266997ce1844a5657c2f1b02eabdR37

I added it there bc all services should always generate a JSON report of the tests inside the container to the host `reports` directory when they run as a generic functionality. Then those JSON reports become our starting point for any downstream hook in the future for parsing, reporting, alerting, etc. 

The CircleCI hook is just one usage of that. And since that is a general orchestration logic need within the services and not just for this specific use of those reports for CircleCI, I've put it where we have common makefile orchestration.

Reply:
sorry missed that...

yeah totally agree, but if we are doing `cucumber-junit` here and our reasoning is "this is something special for cci" then why "mounting" (`-v ``pwd``/reports:/usr/src/app/reports`) is something that we are treating as "common" behavior...

my thinking: if we are doing `cucumber-junit` here then by extension we should do ``-v `pwd`/reports:/usr/src/app/reports`` here also... so ``make -e DOCKER_RUN_OPTS=-v `pwd`/reports:/usr/src/app/reports ci-test-unit``... 

but this is a moo point so nevermind me 😉

Reply:
Allow me to clarify:
1. Generic mechanism _specific to services_ to export JSON of test results. The purpose is for services to have responsibility to report on their in-container test results. Much like any platform providing a "hook", this is for _any_ generic purpose that the service will handle a-la-carte with a one-off pattern.
2. CircleCI uses the generic test results (hook from above) to do CircleCI-specific processing and uploading.

Purpose and responsibility of #1 fits general service-specifc orchestration.
Purpose and responsibility of #2 (the output of #1) fits this specific usage since it is for one single output/platform.
We can have multiple #2s in the future which all build off of the generic pattern provided in #1.

Hope that clarifies the distinguishing responsibilities.

Reply:
yeah i get that 

the only "thing" that i don;t like it this is "forcing" the export of this JSON file in "generic way" to `pwd` inside the #1

i agree that services itself should be exporting to JSON
i agree that circle should be "converting" JSON to "its own" format by itself
i don't agree that JSON should be forcibly "exported" - by the means of the docker volume bind - inside the services _specific_  mechanism; as this export is _specific_ to circle

as you said:
> 
Purpose and responsibility of #1 fits general *service-specifc* orchestration.

so if we will be needing to add "any downstream hook in the future for parsing, reporting, alerting" we will be forced to do this by using `pwd` bind 

but as I've already said, a moo point and more of the brain teaser than really an issue 😉

Reply:
As in we should not implement #1 using docker mount of current directory+reports directory? 

Two questions:
- Why problems arise with binding with `pwd` for this step?
- What do you suggest we do instead of mounting with `pwd` in #1?

Reply:
i think that we should only generate a JSON file inside #1 and use "system" tools to use it...

so my thinking is:
1. service should generate "results" in some common place `/results` (_the hook_)
2. "orchestration" should use _the hook_ in the way *they* want to not in the way they are forced to



> Two questions:
> 
> * Why problems arise with binding with `pwd` for this step?
none that i see right now (we are using only one repo per build so files are separated, but i see the path in which we will be building/testing multiple repos using `-f accounts/Dockerfile`)
> * What do you suggest we do instead of mounting with `pwd` in #1?
move the mount to orb by setting `DOCKER_RUN_OPTS` here instead of there

and once again; this is nothing that i have a strong stance on, just voicing my opinion 😜